determined-ai/determined
# feat: added a new API endpoint to patch master config while cluster is running [DET-9376]

## Description



- Added a new API endpoint to patch master config while it is still running. At this stage only 'log' field in the master config can be changed using this endpoint.
- A new CLI command has been added ``det master config --log --level <log_level> --color
   <on/off>`` that allows the user to change the log level and/or color of the master config while
   the cluster is still running. ``det master config`` can still be used to get master config.
## Test Plan


- Add a debug print statement to a particular endpoint. Hit that endpoint when the log level is set to `info` and make sure that statement is not printed. Then use the API endpoint/CLI Command to change the log level to 'debug' and make sure that debug print statement is printed.
- Use the Get Master Config endpoint to print the master config. Then call the Patch Master Config endpoint and change the log level. Then call the Get Master Config Endpoint again to make sure the change is reflected in the master config.


## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket

DET-9376
